test/framework/src/main/java/org/opensearch/test/InternalTestCluster.java

```diff
@@ -1427,17 +1427,19 @@ private void assertNoSnapshottedIndexCommit() throws Exception {
             for (NodeAndClient nodeAndClient : nodes.values()) {
                 IndicesService indexServices = getInstance(IndicesService.class, nodeAndClient.name);
                 for (IndexService indexService : indexServices) {
-                    for (IndexShard indexShard : indexService) {
-                        try {
-                            Engine engine = IndexShardTestCase.getEngine(indexShard);
-                            if (engine instanceof InternalEngine) {
-                                assertFalse(
-                                    indexShard.routingEntry().toString() + " has unreleased snapshotted index commits",
-                                    EngineTestCase.hasSnapshottedCommits(engine)
-                                );
-                            }
-                        } catch (AlreadyClosedException ignored) {
+                    if (indexService.getIndexSettings().isSegRepEnabled() == false) {
```

Comment:
With SR, the Source Service will still hold a reference to its latest CopyState until it is shut down, so skipped this assertion here as it is made before node shutdown.